pypeit/calibrations.py

```diff
@@ -564,61 +590,59 @@ def get_flats(self):
 
         # Prep
-        trace_image_files, self.master_key_dict['flat'] = self._prep_calibrations('trace')
-
-        # Return cached images
-        if self._cached('flatimages', self.master_key_dict['flat']):
-            self.flatimages = self.calib_dict[self.master_key_dict['flat']]['flatimages']
-            self.flatimages.is_synced(self.slits)
-            self.slits.mask_flats(self.flatimages)
-            return self.flatimages
+        illum_image_files, self.master_key_dict['flat'] = self._prep_calibrations('illumflat')
+        pixflat_image_files, self.master_key_dict['flat'] = self._prep_calibrations('pixelflat')
 
         masterframe_filename = masterframe.construct_file_name(flatfield.FlatImages,
                                                            self.master_key_dict['flat'], master_dir=self.master_dir)
         # The following if-elif-else does:
-        #   1.  Try to load a MasterFrame (if reuse_masters is True)
+        #   1.  Try to load a MasterFrame (if reuse_masters is True).  If successful, pass it back
         #   2.  Build from scratch
-        #   3.  Replace the built pixel-flat with user supplied (e.g. LRISb)
+        #   3.  Load any user-supplied images to over-ride any built
 
+        # Load MasterFrame?
         if os.path.isfile(masterframe_filename) and self.reuse_masters:
-            # Load MasterFrame
             self.flatimages = flatfield.FlatImages.from_file(masterframe_filename)
             self.flatimages.is_synced(self.slits)
             self.slits.mask_flats(self.flatimages)
-        elif len(trace_image_files) > 0:
-            # Process/combine the input pixelflat frames
-            # TODO -- Include an illum frametype eventually
-            # TODO This is incorrect nad is only a hack for LRIS-Blue where we want to only run this code
-            # for the illumination flat construction. There needs to be a pixelflat and an illumflat, and
-            # in cases where the files are different, we need to run the flat field code twice.
-            stacked_traceflat = buildimage.buildimage_fromlist(self.spectrograph, self.det,
-                                                    self.par['traceframe'],
-                                                    trace_image_files,
-                                                    bias=self.msbias, bpm=self.msbpm)
+            return self.flatimages
+
+
+        # Generate the image?
+        #if len(pixflat_image_files) > 0:
+        #    stacked_pixelflat = buildimage.buildimage_fromlist(self.spectrograph, self.det,
+        #                                                       self.par['pixelflatframe'],
+        #                                                       pixflat_image_files,
+        #                                                       bias=self.msbias, bpm=self.msbpm)
+        # TODO -- Allow for separate pixelflat and illumflat images
+        if len(illum_image_files) > 0:
```

Comment:
As far as I understand from the logic here, if the user labels images as pixelflat and illumflat, and if those images are the same, then this code here works as before. However, if the user specifices only pixelflat but not illumflat, the code will not flat field at all. It prints out a warning instead. 

This seems like a big failure mode, in the sense that users may not know they need to include illumination flats. Also if they type use_illumflat = False, the code still won't pixelflat field becuase it never makes the pixel flat becuase illumflats were not set.

Reply:
These cases are all caught by `check_for_calibs()`

Reply:
(and if not the code will crash, as we have decided is best)

Reply:
This does not appear to be the case to me, i.e. if use_illumflat is False, and if the user specifies pixelflats only in the pypeit file, as far as I can see the code does not flat field at all, i.e. it never enters into the flat creation block because len(illumfat_files) = 0

Reply:
Got it.  Modified to allow for either to be provided and if both sets are provided
then we analyze the `illumflat_files` and ignore the `pixelflat_files`

Reply:
See:

https://github.com/pypeit/PypeIt/blob/calibrations_refactor/doc/flat_fielding.rst